backend/fms_core/services/library.py

```diff
@@ -58,16 +78,86 @@ def create_library(library_type, index, platform, strandedness, library_size=Non
 
 def convert_library(process, platform, sample_source, container_destination, coordinates_destination, volume_used,
                     volume_destination, execution_date, comment):
-    library_destination = None
+    sample_destination = None
     errors = []
     warnings = []
 
-    if not process:
-        errors.append(f"Process is required.")
     if not platform:
         errors.append(f"Platform is required.")
     if not sample_source:
         errors.append(f"Source sample is required.")
+    else:
+        for derived_sample_source in sample_source.derived_samples.all():
+            library_source_obj = derived_sample_source.library
+              
+            if library_source_obj.platform == platform:
+                errors.append(f"Source library platform and destination library platform can't be the same.")
+
+    new_library_info = {
+        "platform": platform,
+        "strandedness": SINGLE_STRANDED, # This is only valid as long as MGI is the only conversion acceptable
+    }
+    sample_destination, errors_inherit, warnings_inherit = _inherit_library(process=process,
+                                                                            new_library_info=new_library_info,
+                                                                            sample_source=sample_source,
+                                                                            container_destination=container_destination,
+                                                                            coordinates_destination=coordinates_destination,
+                                                                            volume_used=volume_used,
+                                                                            volume_destination=volume_destination,
+                                                                            execution_date=execution_date,
+                                                                            comment=comment)
+    errors.extend(errors_inherit)
+    warnings.extend(warnings_inherit)
+
+    return sample_destination, errors, warnings
+
+
+def capture_library(process, library_selection, sample_source, container_destination, coordinates_destination, volume_used,
+                    volume_destination, execution_date, comment):
+    sample_destination = None
+    errors = []
+    warnings = []
+
+    if not library_selection:
+        errors.append(f"Capture type is required.")
+
+    if not sample_source:
+        errors.append(f"Source sample is required.")
+    else:
+        for derived_sample_source in sample_source.derived_samples.all():
+            library_source_obj = derived_sample_source.library
+              
+            if library_source_obj.library_selection is not None:
```

Comment:
This will crash because even if you check that the source sample is a library you go on and call this service with the source sample object.